File: config/includes.chroot/usr/essentials/options.py
#!/usr/bin/env python3

import gi
import subprocess

gi.require_version("Gtk", "3.0")
from gi.repository import Gtk, Gdk


class PowerWindow(Gtk.Window):
    def __init__(self):
        super().__init__(title="Power Options")
        self.set_default_size(320, 360)
        self.set_position(Gtk.WindowPosition.CENTER)
        
        self.set_resizable(False)
        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)

        header = Gtk.HeaderBar()
        header.set_show_close_button(True)
        header.set_decoration_layout("close:")
        header.props.title = "System"
        self.set_titlebar(header)

        css = b"""
        window { 
            background-color: #1e1e1e;
            color: #ffffff;
            font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif;
        }
        .header-title {
            font-size: 18px;
            font-weight: 600;
            color: #ffffff;
            margin-bottom: 5px;
        }
        .header-subtitle {
            font-size: 13px;
            color: #98989d;
            margin-bottom: 15px;
        }
        .action-card {
            background-color: rgba(255, 255, 255, 0.06);
            border-radius: 14px;
            padding: 8px;
            border: 1px solid rgba(255, 255, 255, 0.1);
        }
        button.mac-btn {
            background-color: transparent;
            color: #ffffff;
            border-radius: 8px;
            padding: 12px;
            font-weight: 500;
            font-size: 14px;
            border: none;
            box-shadow: none;
        }
        button.mac-btn:hover {
            background-color: rgba(255, 255, 255, 0.1);
        }
        button.btn-danger {
            color: #FF3B30; /* macOS System Red */
            font-weight: 600;
        }
        button.btn-danger:hover {
            background-color: rgba(255, 59, 48, 0.15);
        }
        button.btn-cancel {
            background-color: rgba(255, 255, 255, 0.06);
            color: #0A84FF; /* macOS System Blue */
            font-weight: 600;
            border-radius: 12px;
            margin-top: 10px;
        }
        button.btn-cancel:hover {
            background-color: rgba(255, 255, 255, 0.12);
        }
        """

        provider = Gtk.CssProvider()
        provider.load_from_data(css)
        Gtk.StyleContext.add_provider_for_screen(
            Gdk.Screen.get_default(), provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION
        )

        outer = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=5)
        outer.set_margin_top(20)
        outer.set_margin_bottom(20)
        outer.set_margin_start(30)
        outer.set_margin_end(30)
        self.add(outer)

        title_label = Gtk.Label(label="Power Options")
        title_label.get_style_context().add_class("header-title")
        outer.pack_start(title_label, False, False, 0)

        subtitle_label = Gtk.Label(label="What do you want to do?")
        subtitle_label.get_style_context().add_class("header-subtitle")
        outer.pack_start(subtitle_label, False, False, 0)

        action_card = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=2)
        action_card.get_style_context().add_class("action-card")
        outer.pack_start(action_card, False, False, 10)

        shutdown_btn = Gtk.Button(label="Shut Down")
        shutdown_btn.get_style_context().add_class("mac-btn")
        shutdown_btn.get_style_context().add_class("btn-danger")
        
        restart_btn = Gtk.Button(label="Restart")
        restart_btn.get_style_context().add_class("mac-btn")
        
        logout_btn = Gtk.Button(label="Log Out")
        logout_btn.get_style_context().add_class("mac-btn")

        action_card.pack_start(shutdown_btn, True, True, 0)
        
        action_card.pack_start(Gtk.Separator(), False, False, 0)
        action_card.pack_start(restart_btn, True, True, 0)
        
        action_card.pack_start(Gtk.Separator(), False, False, 0)
        action_card.pack_start(logout_btn, True, True, 0)

        cancel_btn = Gtk.Button(label="Cancel")
        cancel_btn.get_style_context().add_class("mac-btn")
        cancel_btn.get_style_context().add_class("btn-cancel")
        outer.pack_end(cancel_btn, False, False, 0)

        # Event Listeners
        shutdown_btn.connect("clicked", self.shutdown)
        restart_btn.connect("clicked", self.restart)
        logout_btn.connect("clicked", self.logout)
        cancel_btn.connect("clicked", self.cancel)

    def confirm(self, text):
        dialog = Gtk.MessageDialog(
            transient_for=self,
            flags=0,
            message_type=Gtk.MessageType.QUESTION,
            buttons=Gtk.ButtonsType.YES_NO,
            text=text
        )
        response = dialog.run()
        dialog.destroy()
        return response == Gtk.ResponseType.YES

    def shutdown(self, widget):
        if self.confirm("Are you sure you want to shut down your computer now?"):
            subprocess.run(["systemctl", "poweroff"])

    def restart(self, widget):
        if self.confirm("Are you sure you want to restart your computer now?"):
            subprocess.run(["systemctl", "reboot"])

    def logout(self, widget):
        if self.confirm("Are you sure you want to log out from system?"):
            subprocess.run(["openbox", "--exit"])

    def cancel(self, widget):
        Gtk.main_quit()


win = PowerWindow()
win.connect("destroy", Gtk.main_quit)
win.show_all()
Gtk.main()

File: config/includes.chroot/usr/essentials/options.c
#include <gtk/gtk.h>
#include <gdk/gdk.h>
#include <glib.h>
#include <stdio.h>
#include <string.h>

static void load_css(void)
{
    GtkCssProvider *p = gtk_css_provider_new();
    gtk_css_provider_load_from_data(p,
        "window {"
        "  background-color: #1e1e1e;"
        "  color: #ffffff;"
        "  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;"
        "}"
        ".header-title {"
        "  font-size: 18px; font-weight: 600; color: #ffffff; margin-bottom: 5px;"
        "}"
        ".header-subtitle {"
        "  font-size: 13px; color: #98989d; margin-bottom: 15px;"
        "}"
        ".action-card {"
        "  background-color: rgba(255,255,255,0.06);"
        "  border-radius: 14px; padding: 8px;"
        "  border: 1px solid rgba(255,255,255,0.1);"
        "}"
        "button.mac-btn {"
        "  background-color: transparent; color: #ffffff;"
        "  border-radius: 8px; padding: 12px;"
        "  font-weight: 500; font-size: 14px; border: none;"
        "}"
        "button.mac-btn:hover { background-color: rgba(255,255,255,0.1); }"
        ".btn-danger { color: #FF3B30; font-weight: 600; }"
        ".btn-danger:hover { background-color: rgba(255,59,48,0.15); }"
        ".btn-cancel {"
        "  background-color: rgba(255,255,255,0.06); color: #0A84FF;"
        "  font-weight: 600; border-radius: 12px; margin-top: 10px;"
        "}"
        ".btn-cancel:hover { background-color: rgba(255,255,255,0.12); }",
        -1, NULL);
    gtk_style_context_add_provider_for_screen(gdk_screen_get_default(),
        GTK_STYLE_PROVIDER(p), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
    g_object_unref(p);
}

static gboolean confirm(GtkWindow *parent, const char *text)
{
    GtkWidget *dialog = gtk_message_dialog_new(parent,
        GTK_DIALOG_MODAL, GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO, "%s", text);
    gint r = gtk_dialog_run(GTK_DIALOG(dialog));
    gtk_widget_destroy(dialog);
    return r == GTK_RESPONSE_YES;
}

static void on_shutdown(GtkWidget *btn, GtkWindow *parent)
{
    if (confirm(parent, "Are you sure you want to shut down your computer now?"))
        g_spawn_command_line_async("systemctl poweroff", NULL);
}

static void on_restart(GtkWidget *btn, GtkWindow *parent)
{
    if (confirm(parent, "Are you sure you want to restart your computer now?"))
        g_spawn_command_line_async("systemctl reboot", NULL);
}

static void on_logout(GtkWidget *btn, GtkWindow *parent)
{
    if (confirm(parent, "Are you sure you want to log out from system?"))
        g_spawn_command_line_async("openbox --exit", NULL);
}

static void on_cancel(GtkWidget *btn, GtkWindow *win)
{
    gtk_main_quit();
}

int main(int argc, char **argv)
{
    gtk_init(&argc, &argv);
    load_css();

    GtkWidget *win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
    gtk_window_set_title(GTK_WINDOW(win), "Power Options");
    gtk_window_set_default_size(GTK_WINDOW(win), 320, 360);
    gtk_window_set_position(GTK_WINDOW(win), GTK_WIN_POS_CENTER);
    gtk_window_set_resizable(GTK_WINDOW(win), FALSE);
    gtk_window_set_type_hint(GTK_WINDOW(win), GDK_WINDOW_TYPE_HINT_DIALOG);
    g_signal_connect(win, "destroy", G_CALLBACK(gtk_main_quit), NULL);

    GtkWidget *header = gtk_header_bar_new();
    gtk_header_bar_set_show_close_button(GTK_HEADER_BAR(header), TRUE);
    gtk_header_bar_set_decoration_layout(GTK_HEADER_BAR(header), "close:");
    gtk_header_bar_set_title(GTK_HEADER_BAR(header), "System");
    gtk_window_set_titlebar(GTK_WINDOW(win), header);

    GtkWidget *outer = gtk_box_new(GTK_ORIENTATION_VERTICAL, 5);
    gtk_widget_set_margin_top(outer, 20);
    gtk_widget_set_margin_bottom(outer, 20);
    gtk_widget_set_margin_start(outer, 30);
    gtk_widget_set_margin_end(outer, 30);
    gtk_container_add(GTK_CONTAINER(win), outer);

    GtkWidget *title = gtk_label_new("Power Options");
    gtk_style_context_add_class(gtk_widget_get_style_context(title), "header-title");
    gtk_box_pack_start(GTK_BOX(outer), title, FALSE, FALSE, 0);

    GtkWidget *subtitle = gtk_label_new("What do you want to do?");
    gtk_style_context_add_class(gtk_widget_get_style_context(subtitle), "header-subtitle");
    gtk_box_pack_start(GTK_BOX(outer), subtitle, FALSE, FALSE, 0);

    GtkWidget *card = gtk_box_new(GTK_ORIENTATION_VERTICAL, 2);
    gtk_style_context_add_class(gtk_widget_get_style_context(card), "action-card");
    gtk_box_pack_start(GTK_BOX(outer), card, FALSE, FALSE, 10);

    GtkWidget *shutdown_btn = gtk_button_new_with_label("Shut Down");
    gtk_style_context_add_class(gtk_widget_get_style_context(shutdown_btn), "mac-btn");
    gtk_style_context_add_class(gtk_widget_get_style_context(shutdown_btn), "btn-danger");
    g_signal_connect(shutdown_btn, "clicked", G_CALLBACK(on_shutdown), win);
    gtk_box_pack_start(GTK_BOX(card), shutdown_btn, TRUE, TRUE, 0);

    gtk_box_pack_start(GTK_BOX(card), gtk_separator_new(GTK_ORIENTATION_HORIZONTAL), FALSE, FALSE, 0);

    GtkWidget *restart_btn = gtk_button_new_with_label("Restart");
    gtk_style_context_add_class(gtk_widget_get_style_context(restart_btn), "mac-btn");
    g_signal_connect(restart_btn, "clicked", G_CALLBACK(on_restart), win);
    gtk_box_pack_start(GTK_BOX(card), restart_btn, TRUE, TRUE, 0);

    gtk_box_pack_start(GTK_BOX(card), gtk_separator_new(GTK_ORIENTATION_HORIZONTAL), FALSE, FALSE, 0);

    GtkWidget *logout_btn = gtk_button_new_with_label("Log Out");
    gtk_style_context_add_class(gtk_widget_get_style_context(logout_btn), "mac-btn");
    g_signal_connect(logout_btn, "clicked", G_CALLBACK(on_logout), win);
    gtk_box_pack_start(GTK_BOX(card), logout_btn, TRUE, TRUE, 0);

    GtkWidget *cancel_btn = gtk_button_new_with_label("Cancel");
    gtk_style_context_add_class(gtk_widget_get_style_context(cancel_btn), "mac-btn");
    gtk_style_context_add_class(gtk_widget_get_style_context(cancel_btn), "btn-cancel");
    g_signal_connect(cancel_btn, "clicked", G_CALLBACK(on_cancel), win);
    gtk_box_pack_end(GTK_BOX(outer), cancel_btn, FALSE, FALSE, 0);

    gtk_widget_show_all(win);
    gtk_main();
    return 0;
}
